awscli/customizations/cloudformation/artifact_exporter.py

```diff
@@ -659,7 +659,19 @@ def export(self):
 
         self.template_dict = self.export_global_artifacts(self.template_dict)
 
-        for resource_id, resource in self.template_dict["Resources"].items():
+        self.export_resources(self.template_dict["Resources"])
+
+        return self.template_dict
+
+    def export_resources(self, resource_dict):
+        for resource_id, resource in resource_dict.items():
+
+            if resource_id.startswith("Fn::ForEach::"):
+                if not isinstance(resource, list) or len(resource) < 3:
```

Comment:
Can the length of `resource` be greater than 3? My understanding is that the list contains an id, collection, and output dict.

Reply:
The length of `resource` currently can't be greater than 3, but there can be enhancements that could make it larger than 3 in the future, but for now, modified this condition to ensure the size of `resource` is exactly 3